content/docs/azimuth/layer2.md

```diff
@@ -0,0 +1,387 @@
+ultimately up to the roller.
+
+Computing the resulting state transitions obtained from downloaded Ethereum
+transactions is done using `/lib/naive.hoon`, which is a gate that accepts both layer
+1 transaction logs and layer 2 batches, and then computes the resulting state
+transitions and updates the ship's internal Azimuth data held in
+the Gall agent `/app/azimuth.hoon` accordingly.
+
+### Savings in gas costs
+
+There are several dimensions by which naive rollups saves on gas over layer 1.
+They are:
+
+ 1. Gas is not spent on instructing the EVM to compute state transitions and
+    confirm validity.
+ 2. Layer 2 Azimuth state is not stored on Ethereum, so the only data storage
+    gas costs is for the transactions themselves.
+ 3. Layer 2 transactions are written in a highly compressed form. E.g., instead
+    of calling the spawn action by name, it is simply referred to as action `%1`.
+ 4. By collecting multiple layer 2 transactions and submitting them as a single
+    "batch transaction", additional gas savings are achieved by not needing to
+    duplicate information such as which smart contract the transactions are
+    intended for.
+    
+Put together, these create a reduction in gas costs of at least 65x. This is
+true even for batches of L2 transactions that contain only a single transaction.
+Thus there is reason to use one's own ship as an aggregator.
+
+### One way trip
+
+Moving to layer 2 is a one-way trip for now. This means that once a ship moves
+to layer 2, it cannot be moved back to layer 1. We believe it to be technically
+possible to engineer a return trip, and expect that someday this will be the
+case, but there are no plans to implement this in the near future.
+
+## Interacting with L2
+
+Layer 2 ships can perform the same actions on layer 2 that they could on
+layer 1, but are no longer able to perform any layer 1 actions. Layer 1 ships
+can also perform a subset of layer 2 actions - namely the ones related to
+sponsorship.
+
+For a complete list of what layer 2 actions each ship rank, layer, and proxy can
+perform, see [Layer 2 Actions](/docs/azimuth/l2-actions). For an explicit
+description of the byte format of Layer 2 Ethereum transaction, see [Bytestring format](/docs/azimuth/bytestring).
+
+### Sponsorship
+
+Due to the possibility of sponsors and sponsees existing on different layers,
+the precise logic of how sponsorship works is complex. However, under common
+circumstances it is simple.
+
+If either the sponsor or sponsee are on layer 2, then sponsorship actions must
+occur on layer 2. The only exception to this is detaching. A sponsor on layer 1
+may perform a layer 1 detach action on a layer 2 sponsee, and this will result
+in the sponsee having no sponsor on layer 1, and layer 2 as well if they were
+the sponsor on layer 2. This is necessary to simplify the logic, but it also
+guarantees that there is no hard requirement to ever utilize layer 2. Without
```

Comment:
Maybe note the reason we want to not require using layer 2 is some ships are owned by smart contracts, and those can't sign layer 2 transactions.